First pass - Type errors

ostrokach opened the pull request.

ostrokach committed:
E1126: Rephrase error description

ostrokach committed:
E1127: Make the wording more generic since you can also assign to slices

ostrokach committed:
E1131: Add link to python documentation

- For some reason, I do not get these errors with pylint?

ostrokach committed:
E0632: Also applies when there are too many items in sequence

ostrokach committed:
E1135: Change wording to so that it is similar to the next example

ostrokach committed:
E1130: Fix error description so that it corresponds with the example

- Adding string to integer does not trigger this error

ostrokach committed:
E1111: Modify example to demonstrate the problem

ostrokach committed:
E1128: Add a more concrete example

david-yz-liu commented on index.md:
```diff
@@ -106,14 +106,14 @@ a value which is an integer, not a function.
-f() does not return anything. As a result, `x` always gets the value `None`.
+This error occurs when you assign a variable to the return value of a function call, but the function never returns anything. This error is similar to [E1128](#E1128).
+
+In the following example, `add_fruit` modifies `fruit_basket` inplace and does not return anything. As a result, `new_fruit_basket` always gets the value `None`.
```
Use the term "mutate" rather than "modify," and don't use the term "inplace."

david-yz-liu commented on index.md:
```diff
@@ -106,14 +106,14 @@ a value which is an integer, not a function.
 ~~~~ {include="E1111_assignment_from_no_return"}
 ~~~~
 
+You should either modify `add_fruit` to return a new list, or call `add_fruit` without assigning the return value to a variable.
```
Let's use the first-person voice for these examples: "We should either..."

david-yz-liu reviewed (changes requested):
Great work. I only left a few minor comments.

ostrokach committed:
E1111: Fixup the previous commit

- Use the term "mutate" instead of "modify"
- Don't use the term "inplace"
- Use first-person voice

nigef commented on index.md:
```diff
@@ -187,41 +187,36 @@ print(a[0:3])
 
-This error occurs when you use the membership test `a in b`, but the `b`'s type
-does not support this type of check.
+This error occurs when you use the membership test `a in b`, but the type of `b`
```
Please avoid using the word "you", here and in other instances.  I believe this has been mentioned before.
There are many ways a phrase could be reworded, for example, 
> This error occurs when the membership of `a` is tested in `b`, i.e.  `a in b`, but the type of `b` does not support membership tests.

nigef commented on index.md:
```diff
@@ -1343,27 +1338,27 @@ print 3   # Error on this line
+
     e. There is no ++ increment or –- decrement operator. Do not try to increment
-    or decrement a variable with ++ or --.
+    or decrement a variable with ++ or --.~~~~
```
typo?

nigef commented on index.md:
```diff
@@ -106,14 +106,14 @@ a value which is an integer, not a function.
 ~~~~ {include="E1111_assignment_from_no_return"}
 ~~~~
 
+We should either modify `add_fruit` to return a new list, or call `add_fruit` without assigning the return value to a variable.
```
Please reword to not use "we".
